library_conventions/packages.adoc

```diff
@@ -0,0 +1,232 @@
+https://github.com/LibrePCB/librepcb-doc/issues?q=is%3Aissue+label%3A%22Conventions%3A+Packages%22[on GitHub]!
+====
+
+
+[#libraryconventions-packages-scope]
+==== Scope
+
+The most important thing to consider when creating a Package is the scope
+of it. Since LibrePCB handles footprints different than other EDA tools,
+special attention is required here.
+
+Actually it's quite simple: Just imagine how the part looks exactly (the
+mechanical shape, dimension and color). If two parts look exactly (or _almost_)
+equal, they can use the same Package. If they look different, two separate
+Packages must be created.
+
+[TIP]
+====
+*Don't think about the land pattern (i.e. footprint) of the part* -- it's not
+relevant for this decision. Even if a package can be mount differently on
+a PCB (e.g. a THT resistor can be mount horizontally or vertically) and thus
+require different footprints, only one Package is needed.
+====
+
+.Color (e.g. 0805 LED)
+====
+Even if a red 0805 LED has exactly the same footprint as a blue 0805 LED,
+they should have *separate Packages because of the different color*. This way
+a Device can link to the Package with the proper color, and thus it will appear
+with the proper color in the 3D PCB preview (once LibrePCB supports 3D models).
+====
+
+.Height (e.g. SO-8)
+====
+Some packages are available with different heights, e.g. SO-8 is available with
+heights of 1.2mm and 1.4mm. *As the 3D models would be different, separate
+Packages are needed*.
+
+_Note: To avoid creating too many Packages, a small tolerance is allowed. So for
+a device with a height of 1.3mm you can use the Package with a height of 1.4mm._
+====
+
+.Mounting variants (e.g. TO220)
+====
+Many packages can be mount either vertically or horizontally, for example the
+TO220. If mounted horizontally, there might be a hole in the PCB to screw the
+metal tab down to the PCB, or you may want to solder the tab to the PCB without
+a hole in it. For all these cases *only one Package is needed* -- the different
+mounting variants should be handled by different footprints inside the Package.
+====
+
+
+[#libraryconventions-packages-naming]
+==== Naming
+
+Following conventions apply to Package names:
+
+* *Language must be American English* (en_US), if applicable (many Packages
+  have language-neutral names anyway).
+* *If {ipc7351-naming-pdf}[IPC-7351] specifies a name for a given package, use
+  that name* (e.g. "SOT95P280X145-5" instead of "SOT23-5"). Alternative names
+  (like "SOT23-5") should be added to the comma-separated keywords list.
+* *Size information must be given in millimeters*, not inches or mils. For
+  packages which are well known by their size in inches (e.g. "0805" which is
+  "2012" in metric), it's recommended to write the well known name in
+  parentheses. For example a resistor could be called "RESC2012X70 (0805)".
+* For packages which are available with different pin counts, *append the pin
```

Comment:
Hmm good point. Should we just use SOIC as example? :D

> It could be relevant in the description though, so we could reword the item like this:

I don't understand, what do you mean exactly?

Reply:
I refactored the whole section a bit. Using a hyphen as separator now only applies for packages not covered by IPC. I kept the example with DIP-8 as it is very well known, and in the end it only demonstrates the use of the hyphen...